apps/fm/test/unit/fm_test.js

```diff
@@ -70,6 +83,72 @@ suite('FM', function() {
       assert.equal($('frequency').textContent, 87.5);
     });
 
+    test('compare new set frequency with aria-valuenow', function() {
+      var frequency = 95.7;
+      var freq = frequencyDialer.setFrequency(frequency);
+      assert.equal(parseFloat(
+        $('dialer-container').getAttribute('aria-valuenow')), frequency);
+    });
+
+    test('screen reader swipe up on dialer should advance 0.1 Hz', function() {
+      var setFreq = frequencyDialer.setFrequency(95.7);
+
+      var keyEvent = document.createEvent('KeyboardEvent');
+      keyEvent.initKeyEvent('keypress',
+        true, true, window,
+        false, false, false, false,
+        38, 0); // DOM_VK_UP = 38
+
+      $('dialer-container').dispatchEvent(keyEvent);
+      assert.equal(frequencyDialer._currentFreqency, 95.8);
+    });
+
+    test('screen reader swipe down on dialer should reverse 0.1 Hz',
+      function() {
+        var setFreq = frequencyDialer.setFrequency(95.7);
+
+        var keyEvent = document.createEvent('KeyboardEvent');
+        keyEvent.initKeyEvent('keypress',
+          true, true, window,
+          false, false, false, false,
+          40, 0); // DOM_VK_DOWN = 40
+
+        $('dialer-container').dispatchEvent(keyEvent);
+        assert.equal(frequencyDialer._currentFreqency, 95.6);
+      }
+    );
+
+    test('maintain aria-pressed on bookmark button', function() {
+      var bb = $('bookmark-button');
+
+      // Start at arbitrary station
+      mozFMRadio.setFrequency(88.6);
```

Comment:
I don't think that this is right...

The setFrequency() wrapper function only invokes the frequencyDialer.setFrequency() function, which itself never touches the mozFMRadio object.  So, when we call updateFreqUI() -- which is where the bookmark button's aria-pressed attribute is modified -- it looks to find the mozFMRadio frequency in the FavoritesList and can't find it, because it hadn't been set appropriately.